data-serving/data-service/src/controllers/case.ts

```diff
@@ -105,39 +112,105 @@ export class CasesController {
             const casesIgnoringExcluded = this.excludeRestrictedSourcesFromCaseAggregation(
                 casesQuery,
             );
-            const matchingCases = await Case.aggregate(
-                casesIgnoringExcluded,
-            ).collation({
-                locale: 'en_US',
-                strength: 2,
-            });
 
-            res.setHeader('Content-Type', 'text/csv');
-            res.setHeader(
-                'Content-Disposition',
-                'attachment; filename="cases.csv"',
-            );
+            let matchingCases: any;
+            if (queryLimit) {
+                matchingCases = await Case.aggregate(casesIgnoringExcluded)
+                    .collation({
+                        locale: 'en_US',
+                        strength: 2,
+                    })
+                    .limit(queryLimit);
+            } else {
+                matchingCases = await Case.aggregate(
+                    casesIgnoringExcluded,
+                ).collation({
+                    locale: 'en_US',
+                    strength: 2,
+                });
+            }
```

Comment:
Not a big deal, but could be changed to:

```
let query:any = Case.aggregate(casesIgnoringExcluded)collation({locale: 'en_US', strength: 2, });
if (queryLimit) { query = query.limit(queryLimit); }
const matchingCases = await query;
```

Reply:
Yeah, the code will be cleaner, thank you for spotting that!